### PR TITLE
✨  add diagnostic script for CI log collection

### LINKED ISSUE #474: Diagnostic script for log collection
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
An easy to use script that will collect logs for kubernetes, iptables, instance metadata, etc.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by chewong**
**What this PR does / why we need it**:

This PR adds a diagnostic script for CI log collection. I also added a script called `redact.sh` which will redact all sensitive information for all log files like:

```
        {
          "cloud": "AzurePublicCloud",
          "tenantId": "===REDACTED===",
          "subscriptionId": "===REDACTED===",
          "aadClientId": "===REDACTED===",
          "aadClientSecret": "===REDACTED===",
          "resourceGroup": "capz-h9gtqc",
          "securityGroupName": "capz-h9gtqc-node-nsg",
          "location": "southcentralus",
          "vmType": "vmss",
          "vnetName": "capz-h9gtqc-vnet",
          "vnetResourceGroup": "capz-h9gtqc",
          "subnetName": "capz-h9gtqc-node-subnet",
          "routeTableName": "capz-h9gtqc-node-routetable",
          "loadBalancerSku": "standard",
          "maximumLoadBalancerRuleCount": 250,
          "useManagedIdentityExtension": false,
          "useInstanceMetadata": true
        }
```

The file structure of the artifacts folder:
```
_artifacts
├── management-cluster
│   ├── capz-control-plane
│   │   ├── alternatives.log
│   │   ├── containerd.log
│   │   ├── containers
│   │   │   ├── <container log>
│   │   ├── pods
│   │   │   ├── <pod log>
│   │   └── serial.log
│   ├── docker-info.txt
│   ├── images.info
│   ├── kind-cluster.info
│   └── resources
│       ├── azureclusters.log
│       ├── azuremachinepools.log
│       ├── azuremachines.log
│       ├── azuremachinetemplates.log
│       ├── clusters.log
│       ├── kubeadmconfigs.log
│       ├── kubeadmconfigtemplates.log
│       ├── kubeadmcontrolplanes.log
│       ├── machinedeployments.log
│       ├── machinepools.log
│       ├── machines.log
│       └── machinesets.log
└── workload-cluster
    ├── capz-5fk43b-control-plane-llp96
    │   ├── cloud-init-output.log
    │   ├── cloud-init.log
    │   ├── containerd.log
    │   ├── journal.log
    │   ├── kern.log
    │   ├── kubelet.log
    │   └── pods
    │   │   ├── <pod log>
    │   ├── kern.log
    │   ├── kubelet.log
...
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes #534
Fixes #474

/test pull-cluster-api-provider-azure-conformance-v1alpha3
/test pull-cluster-api-provider-azure-conformance-machine-pool-v1alpha3

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
✨  diagnostic script for log collection
```

### CODE REVIEW

**Comment by chewong** [comment]
/hold
waiting for #653 to get merged first.

File: hack/log/redact.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+# Verify the required Environment Variables are present.
+: "${AZURE_SUBSCRIPTION_ID:?Environment variable empty or not defined.}"
+: "${AZURE_TENANT_ID:?Environment variable empty or not defined.}"
+: "${AZURE_CLIENT_ID:?Environment variable empty or not defined.}"
+: "${AZURE_CLIENT_SECRET:?Environment variable empty or not defined.}"
+
+echo "================ REDACTING LOGS ================"
+log_files=( $(find "${ARTIFACTS:-${PWD}/_artifacts}" -type f) )
+# TODO(chewong): add ${AZURE_STANDARD_JSON_B64} and ${AZURE_VMSS_JSON_B64}
```
Comment (CecileRobertMichon) [inline hack/log/redact.sh:29]: @chewong I would do this now if I were you to make sure the PR doesn't accidentally run CI on a rebased version that has the new vars merged

**Comment by chewong** [comment]
> Looks great. I don't see any `${ARTIFACTS}/workload-cluster/` folder in the current run is that expected?

The folder is available in `pull-cluster-api-provider-azure-conformance-v1alpha3` and `pull-cluster-api-provider-azure-conformance-machine-pool-v1alpha3`.

**Pull request merged**